interview-questions.md

```diff
@@ -9,41 +9,42 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: Using 'this' is a way to call on a key value in a component. It is used at the beginning of the key value and followed by dot notation and the value you are trying to call.
 
-  Researched answer:
+  Researched answer: The keyword 'this' refers to the object it belongs to. It has different values depending on where it is used.
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a software designed by Facebook that makes it possible for users to interact with a virtual DOM without
+  the page reloading every time an update is made. React accomplishes this by using nodes on ends of code to create a layer that only "reacts" to the specific changes being made.
 
-  Researched answer:
+  Researched answer: React is a front-end, open-source library used to build UI. It's especially used when building a single page app. You would want to use it because it allows developers to create large apps that can change data without reloading the page. It's quick and simple.
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: The required lifecycle method is render() because it's responsible for getting everything to the UI.
 
-  Researched answer:
+  Researched answer: Render is the only lifecycle method that is required because it handles the rendering of the component to the UI. It happens during the mounting and updating of your component.
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is a combination of JavaScript and HTML. A notable difference between HTML and JSX is that JSX contains logic.
 
-  Researched answer:
+  Researched answer: JAX is a markup that looks like HTML, but gets transformed into JavaScript. Logic is passed through using HTML tags. It provides lass coding and gives more functionality.
```

Comment:
Both of these answers together make a great answer! Highly reccommend you remember this for interviews